File: airbyte-oauth/src/main/java/io/airbyte/oauth/flows/facebook/InstagramOAuthFlow.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.oauth.flows.facebook;

import com.google.common.annotations.VisibleForTesting;
import java.net.http.HttpClient;
import java.util.function.Supplier;

/**
 * Instagram OAuth.
 */
// Instagram Graph API require Facebook API User token
public class InstagramOAuthFlow extends FacebookOAuthFlow {

  private static final String SCOPES =
      "ads_management,business_management,instagram_basic,instagram_manage_insights,pages_show_list,pages_read_engagement";

  public InstagramOAuthFlow(final HttpClient httpClient) {
    super(httpClient);
  }

  @VisibleForTesting
  InstagramOAuthFlow(final HttpClient httpClient, final Supplier<String> stateSupplier) {
    super(httpClient, stateSupplier);
  }

  @Override
  protected String getScopes() {
    return SCOPES;
  }

}

File: airbyte-oauth/src/main/kotlin/io/airbyte/oauth/flows/facebook/InstagramOAuthFlow.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.oauth.flows.facebook

import com.google.common.annotations.VisibleForTesting
import java.net.http.HttpClient
import java.util.function.Supplier

/**
 * Instagram OAuth.
 *
 * Instagram Graph API require Facebook API User token
 */

class InstagramOAuthFlow : FacebookOAuthFlow {
  constructor(httpClient: HttpClient) : super(httpClient)

  @VisibleForTesting
  internal constructor(httpClient: HttpClient, stateSupplier: Supplier<String>) : super(httpClient, stateSupplier)

  override fun getScopes(): String = SCOPES

  companion object {
    private const val SCOPES =
      "ads_management,business_management,instagram_basic,instagram_manage_insights,pages_show_list,pages_read_engagement"
  }
}
